onchain/src/social/request.cairo

```diff
@@ -37,47 +36,189 @@ pub fn verify<C, +Display<C>>(request: @SocialRequest<C>) -> bool {
             sig: Signature {
-                r: 0xfbbe47a42df4b01348982882bb0f622f8eb6802d5c6d9d8779b773a22753e2cc_u256,
-                s: 0x14058232c01aad1f90ee1f34cf8b0e5d0678435890a8bdcfda8f361908486d48_u256
+                r: 0x4fda18c929f820f7f93f310f4fa9a8f2efcdd544539f4ce24fe2daf4f68d0b2d_u256,
+                s: 0x279537893013f5849a716ac48e89ab4f8ce94871986326494c7311fc956639c3_u256
             }
         };
+
         assert!(verify(@r));
     }
+
+    #[test]
+    fn verify_03() {
+        let r: SocialRequest<ByteArray> = SocialRequest {
+            pubkey: 0xcbddbb8b79e395d6458b49aa315b74fcc26d4d8d722e0a6421b4e04a612fc51c_u256,
+            created_at: 1716285235_u64,
+            kind: 1_u16,
+            tags: "[[\"e\",\"5c83da77af1dec6d7289834998ad7aafbd9e2191396d75ec3cc27f5a77226f36\"]]",
+            content: "joyboy",
+            sig: Signature {
+                r: 0x206e086fe298bf0733b0b22316721636ae7d8ce025c76baf83b8a31efaec8821_u256,
+                s: 0x494452ba56fd465a0d69baa1ff4af9efcb1d0af8f107473ce33877d7a1034a8e_u256
+            }
+        };
+
+        assert!(verify(@r));
+    }
+
+    #[test]
+    fn verify_04() {
+        // invalid tags
+        // valid tags =
+        // "[[\"e\",\"5c83da77af1dec6d7289834998ad7aafbd9e2191396d75ec3cc27f5a77226f36\"]]"
+        let r: SocialRequest<ByteArray> = SocialRequest {
+            pubkey: 0xcbddbb8b79e395d6458b49aa315b74fcc26d4d8d722e0a6421b4e04a612fc51c_u256,
+            created_at: 1716285235_u64,
+            kind: 1_u16,
+            tags: "[]",
+            content: "joyboy",
+            sig: Signature {
+                r: 0x206e086fe298bf0733b0b22316721636ae7d8ce025c76baf83b8a31efaec8821_u256,
+                s: 0x494452ba56fd465a0d69baa1ff4af9efcb1d0af8f107473ce33877d7a1034a8e_u256
+            }
+        };
+
+        assert_eq!(verify(@r), false);
+    }
+
+    #[test]
+    fn verify_05() {
+        // invalid content
+        // valid content =
+        // "nprofile1qys8wumn8ghj7un9d3shjtn2daukymme9e3k7mtdw4hxjare9e3k7mgqyzzxqw6wxqyyqqmv4rxgz2l0ej8zgrqfkuupycuatnwcannad6ayqx7zdcy
+        // send 1 USDC to nprofile1qqs2sa3zk4a49umxg4lgvlsaenrqaf33ejkffd78f2cgy4xy38h393s2w22mm"
+        let r: SocialRequest<ByteArray> = SocialRequest {
+            pubkey: 0xa2611fdbcbcc1e43ef809341ddef4a98c15ff6e6410ff7ed0c2b1c4f2a2cc2f5_u256,
+            created_at: 1716403778_u64,
+            kind: 1_u16,
+            tags: "[]",
+            content: "",
+            sig: Signature {
+                r: 0x4fda18c929f820f7f93f310f4fa9a8f2efcdd544539f4ce24fe2daf4f68d0b2d_u256,
+                s: 0x279537893013f5849a716ac48e89ab4f8ce94871986326494c7311fc956639c3_u256
+            }
+        };
+
+        assert_eq!(verify(@r), false);
+    }
+
+    #[test]
+    fn verify_06() {
```

Comment:
change name to `verify_invalid_public_key`, apply the same rule for subsequent tests.